src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -678,6 +678,12 @@ def num_approved_reviews(self):
         return self.reviews.filter(
             status=self.reviews.model.APPROVED).count()
 
+    @property
+    def sorted_recommended_products(self):
+        """Keeping order by recommendation ranking."""
+
+        return [r.recommendation for r in self.primary_recommendations.all()]
```

Comment:
This needs to be `self.primary_recommendations.select_related('recommendation').all()` I think, otherwise additional SQL queries are executed.

Also the empty line should be remove